include/vcpkg/base/optional.h

```diff
@@ -266,151 +256,142 @@ namespace vcpkg
                 return *m_t;
             }
 
+            void destroy() { m_t = nullptr; }
+
         private:
             const T* m_t;
         };
     }
 
     template<class T>
-    struct Optional
+    struct Optional : private details::OptionalStorage<T>
     {
-    private:
-        details::OptionalStorage<T> m_base;
-
     public:
         constexpr Optional() noexcept { }
 
         // Constructors are intentionally implicit
         constexpr Optional(NullOpt) { }
 
-        template<class U, class = std::enable_if_t<!std::is_same_v<std::decay_t<U>, Optional>>>
-        constexpr Optional(U&& t) : m_base(std::forward<U>(t))
+        template<class U,
+                 std::enable_if_t<!std::is_same_v<std::decay_t<U>, Optional> &&
+                                      std::is_constructible_v<details::OptionalStorage<T>, U>,
```

Comment:
Couple notes:

As I said on the call I'm still a bit concerned that this creates unexpected funny conversions, but luckily I think they all will result in compilation failures rather than bad output. It may be possible to use inherited constructors (https://en.cppreference.com/w/cpp/language/using_declaration#Inheriting_constructors) for all this too.

Reply:
I had a version that used inheriting constructors, but I had less confidence in the (more complex) change.